text/0010-exception-types.md

```diff
@@ -0,0 +1,115 @@
+
+A user of Sentry should be able to 
+
+- filter events on the issues page or discover for the newly introduces exception types.
+- highlight (similar to the unhandled label) events of the type unhandled and process termination.
+- get alerted for events of the type unhandled and process termination separately.
+
+Currently, there's an unhandled label on the issue's page but it's only highlighted for process termination errors.
+
+In order to propagate those exception types, the exception mechanism needs to be adapted:
+
+```json
+{
+  "exception": {
+    "values": [
+      {
+        "type": "Error",
+        "value": "An error occurred",
+        "mechanism": {
+          "type": "generic",
+          "handled": true,
+          "process_terminated": false // <--- newly introduced field
+        }
+      }
+    ]
+  }
+}
+```
+
+In order to achieve backwards compatibility, in the absence of the `process_termination` flag, the current behavior stays as is.
+As soon as the `process_terminated` flag is present the bavior is as follows:
+
+- `handled = true` and `process_terminated = true`: Software was gracefully shut down after an handled exception. This should never happen and is invalid.
+- `handled = false` and `process_terminated = false`: Exception is not handled by the user but didn't cause the software to terminate. Same as `unhandled` in the list above
+- `handled = false` and `process_terminated = true`: Software terminated after an unhandled exception. Same as `process termination` in the list above
+- `handled = true` and `process_terminated = false`: Exception was reported via `Sentry.capture*()` method. Same as `handled` in the list above.
+
+In the absence of the `handled` or its value being null, it's assumed to be `handled = true`. This is also the current behavior.
+
+The introduction of the `process_terminated` flag enables the consideration of such exception types in the `session health` metric.
+
+# Option 2
+
+This one is very similar to option 1, however instead of an additional flag, this introduces an enum for the different types.
+Once again, the mechanism needs to be adapted:
+
+```json
+{
+  "exception": {
+    "values": [
+      {
+        "type": "Error",
+        "value": "An error occurred",
+        "mechanism": {
+          "type": "generic",
+          "exception_type": "handled|unhandled|process_termination", // <--- newly introduced field
+        }
+      }
+    ]
+  }
+}
+```
+
+If the currently available `handled` flag is also present, the `exception_type` flag takes precedence. The `handled` flag however should become deprecated.
+
+The introduction of the `exception_type` flag enables the consideration of such exception types in the `session health` metric.
```

Comment:
The release health protocol would need to change as well, because right now we only have `crashed` and `errored`
https://docs.sentry.io/product/releases/health/#session-status
but we need to represent 3 states.

Reply:
It seems to me we don't really need to change Sessions. We can just go from errroed to crashed in case process_exit=true instead of handled:false